src/mixin/pending-container.js

```diff
@@ -0,0 +1,57 @@
+
+export const PendingContainer = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			hasPendingChildren: {
+				type: Boolean
+			},
+			pendingCount: {
+				type: Number
+			},
+		};
+	}
+
+	constructor() {
+		super();
+		this.pendingCount = 0;
+		this.hasPendingChildren = false;
+	}
+
+	connectedCallback() {
+		this.addEventListener('pending-state', this.__onPendingState);
+		if (super.connectedCallback) {
+			super.connectedCallback();
+		}
+	}
+
+	disconnectedCallback() {
+		this.addEventListener('pending-state', this.__onPendingState);
```

Comment:
Looks like this should be `removeEventListener`.

However, if this was rendering an element, the connectedCallback and disconnectedCallback could be avoided.

```
render() {
   return html`<div @d2l-pending-state="${...}"><slot></slot></div>`;
}
```

Reply:
Since this is actually a mixin, no need to make the `render` more complex.